main/tests/server/src/com/google/refine/operations/row/RowKeepMatchedOperationTest.java

```diff
@@ -0,0 +1,204 @@
+
+package com.google.refine.operations.row;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Properties;
+
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import com.google.refine.RefineTest;
+import com.google.refine.browsing.DecoratedValue;
+import com.google.refine.browsing.Engine;
+import com.google.refine.browsing.EngineConfig;
+import com.google.refine.browsing.RowVisitor;
+import com.google.refine.browsing.facets.ListFacet.ListFacetConfig;
+import com.google.refine.expr.MetaParser;
+import com.google.refine.expr.functions.FacetCount;
+import com.google.refine.grel.Function;
+import com.google.refine.grel.Parser;
+import com.google.refine.history.HistoryEntry;
+import com.google.refine.model.Cell;
+import com.google.refine.model.ModelException;
+import com.google.refine.model.Project;
+import com.google.refine.model.Row;
+import com.google.refine.operations.EngineDependentOperation;
+import com.google.refine.operations.OperationRegistry;
+import com.google.refine.util.ParsingUtilities;
+import com.google.refine.util.TestUtils;
+
+public class RowKeepMatchedOperationTest extends RefineTest {
+
+    static final String ENGINE_JSON_DUPLICATES = "{\"facets\":[{\"type\":\"list\",\"name\":\"facet A\",\"columnName\":\"Column A\",\"expression\":\"facetCount(value, 'value', 'Column A') > 1\",\"omitBlank\":false,\"omitError\":false,\"selection\":[{\"v\":{\"v\":true,\"l\":\"true\"}}],\"selectBlank\":false,\"selectError\":false,\"invert\":false}],\"mode\":\"row-based\"}}";
+
+    @BeforeSuite
+    public void registerOperation() {
+        OperationRegistry.registerOperation(getCoreModule(), "row-keep-matched", RowKeepMatchedOperation.class);
+    }
+
+    @Override
+    @BeforeTest
+    public void init() {
+        logger = LoggerFactory.getLogger(this.getClass());
+    }
+
+    // dependencies
+    Project projectIssue567;
+    Properties options;
+    EngineConfig engine_config;
+    Engine engine;
+    Properties bindings;
+
+    Project project;
+    ListFacetConfig facet;
+
+    @Test
+    public void serializeRowKeepMatchedOperation() throws IOException {
+        String json = "{"
+                + "\"op\":\"core/row-keep-matched\","
+                + "\"engineConfig\":{\"facets\":[],\"mode\":\"row-based\"},"
+                + "\"description\":\"Keep only matching rows\""
```

Comment:
Once you have localized the operation description, this will need adapting because of #7058 (let me merge it already).